lib/LanguagePicker.js

```diff
@@ -76,43 +76,48 @@ LanguagePicker.prototype.newProcessor = function newProcessor() {
       }
 
       // 2. Fallback language from fallbacks.json
-      for (let i = 0; i < this.fallbacks.length; i++) {
-        if (this.values[this.fallbacks[i]]) {
-          return this.values[this.fallbacks[i]];
+      for (const fallback of this.fallbacks) {
+        val = this.values[fallback];
+        if (val) {
+          return val;
         }
       }
 
       // 3. Any language with suffix of same script as requested language
-      for (let i = 0; i < valueLangCodes.length; i++) {
-        if (valueLangCodes[i].endsWith(this.langScript)) {
-          return this.values[valueLangCodes[i]];
+      const valueLangCodes = Object.keys(this.values);
+      for (const langCode of valueLangCodes) {
+        val = this.values[langCode];
+        if (val && langCode.endsWith(this.langScript)) {
+          return val;
         }
       }
 
       // 4. Another language with the same script
-      for (let i = 0; i < this.languagesInScript.length; i++) {
-        if (this.values[this.languagesInScript[i]]) {
-          return this.values[this.languagesInScript[i]];
+      for (const langCode of this.languagesInScript) {
+        val = this.values[langCode];
+        if (val) {
+          return val;
         }
       }
 
       // 5. English
-      if (this.values[this.prefix ? `${this.prefix}en` : 'en']) {
-        return this.values[this.prefix ? `${this.prefix}en` : 'en'];
+      val = this.values[this.prefix ? `${this.prefix}en` : 'en'];
+      if (val) {
+        return val;
       }
 
       // 6. Any language with suffix 'Latn'
       // Only do this if the script isn't already latin;
       // otherwise, we've done it in step 3
-      for (let i = 0; i < valueLangCodes.length; i++) {
-        if (valueLangCodes[i].endsWith('Latn')) {
-          return this.values[valueLangCodes[i]];
+      for (const value of valueLangCodes) {
+        if (value && value.endsWith('Latn')) {
+          return this.values[value];
         }
       }
 
       // If nothing is found:
       // - If there's a name tag, return it
-      // - IF there's no name tag, return first choice
+      // - If there's no name tag, return first choice
       return this.nameTag && this.values[this.nameTag] ?
```

Comment:
use `val` too, avoids double lookup.  There might be a JS engine optimization for it, but I don't know if its there.